Sources/BeeTracking/TrackingLikelihoodModel.swift

```diff
@@ -56,6 +56,19 @@ public struct TrackingLikelihoodModel<Encoder: AppearanceModelEncoder, FG:Genera
               foregroundModel : FG(from: fgFeatures, given:p?.foregroundModel),
               backgroundModel : BG(from: bgFeatures, given:p?.backgroundModel))
   }
+  
+  /// Calculate the negative log likelihood of the likelihood model of a patch
+  @differentiable
+  public func negativeLogLikelihood(of patch: Tensor<Double>) -> Double {
+    let encoded = encoder.encode(patch.expandingShape(at: 0)).squeezingShape(at: 0)
```

Comment:
Works in batch?

Reply:
@dellaert I am thinking about changing this interface from `encode()` to `encode(batch: )` and `encode(sample: )`, how is that?

Reply:
done